internal/controller/spinapp_controller.go

```diff
@@ -435,6 +437,41 @@ func constructDeployment(ctx context.Context, app *spinv1alpha1.SpinApp, config
 
 	labels := constructAppLabels(app)
 
+	var container corev1.Container
+	if config.RuntimeClassName != nil {
+		container = corev1.Container{
+			Name:    app.Name,
+			Image:   app.Spec.Image,
+			Command: []string{"/"},
+			Ports: []corev1.ContainerPort{{
+				Name:          spinapp.HTTPPortName,
+				ContainerPort: spinapp.DefaultHTTPPort,
+			}},
+			Env:            env,
+			VolumeMounts:   volumeMounts,
+			Resources:      resources,
+			LivenessProbe:  livenessProbe,
+			ReadinessProbe: readinessProbe,
+		}
+	} else if config.SpinImage != nil {
+		container = corev1.Container{
+			Name:  app.Name,
+			Image: *config.SpinImage,
+			Args:  []string{"up", "--listen", fmt.Sprintf("0.0.0.0:%d", spinapp.DefaultHTTPPort), "-f", app.Spec.Image},
```

Comment:
We _might_ need to include RuntimeConfig somewhere here? - I don't remember how spin picks up runtimeConfig 😅  (a follow up issue to include it in the E2E for spintainers and punting it for now would be fine)

Reply:
Followup issue filed. Good catch. https://github.com/spinkube/spin-operator/issues/319